kernel/util/printer.c

```diff
@@ -0,0 +1,72 @@
+#ifndef TABS_PRINTER_C
```

Comment:
Same here: `UTIL_PRINTER_C`

Reply:
Also, the name printer makes me think this is an actual printer, maybe something like `formatter` could be better? This can also go directly in `terminal.c` in my opinion.

Reply:
It indeed does formatting but as long as it also does terminal printing I would not name it `formatter` as that is not it's current functionality.
In our current kernel it basically is a printer, just a screen printer.

In the future this can definitely be extended to also print to other places using some abstraction layer.